docs/02-customize-a-pizza.md

```diff
@@ -282,11 +290,14 @@ void CancelConfigurePizzaDialog()
 {
     configuringPizza = null;
     showingConfigureDialog = false;
-    StateHasChanged();
 }
 ```
 
-The `StateHasChanged` method signals to the runtime that the component's state has changed and it needs to be rendered. Components are rendered automatically by the runtime when its parameters change or when a UI event is fired on that component. In this case the event triggering the state change came from a different component, so `StateHasChanged` needs to be called manually.
+Now, what happens when you click the dialog cancel button is that `Index.CancelConfigurePizzaDialog` will execute, and then the `Index` component will render itself. Since `showingConfigureDialog` is now `false` the dialog will not be displayed. 
+
+Normally what happens when you trigger an event (like clicking the cancel button) is that the component that defined the event handler delegate will rerender. You could define events using any delegate type like `Action` or `Func<string, Task>`. Sometimes you want to use an event handler delegate that doesn't belong to a component - if you used a normal delegate type to define the event then nothing will be rendered or updated. 
```

Comment:
>  the component that defined the event handler delegate will rerender

That might be misleading. I'd suggest something like "Normally, the framework considers the event as being handled by whichever component rendered the DOM element that received the event, and so that is which component gets re-rendered."

Reply:
I'm confusing myself now. Please ignore my outdated comment above.